skypalwarmup/JSUserRegistrationClient/src/main/java/application/service/RestUserRegistrationConsumerImpl.java

```diff
@@ -0,0 +1,43 @@
+package application.service;
+
+import application.messaging.dto.UserInputDTO;
+import java.net.URI;
+import java.net.URISyntaxException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.MediaType;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Copyright (c) 2016-2018, Jumia.
+ */
+
+@Slf4j
+@Service
+public class RestUserRegistrationConsumerImpl implements RestConsumer {
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Bean
+    public RestTemplate restTemplate() {
+        return new RestTemplate();
+    }
+
+    public boolean postUserRegistration(UserInputDTO userInputDTO) {
+        try {
+            ResponseEntity<String> response = restTemplate.exchange(RequestEntity
+                .post(new URI("http://localhost:8080/api/users"))
```

Comment:
Line 34: This is fine but, of course in a real application this would have to go to the configuration file.